.github/workflows/release.yaml

```diff
@@ -0,0 +1,109 @@
+name: Release
+
+on:
+  workflow_dispatch:
+
+env:
+  RUST_VERSION: 1.68
+
+jobs:
+  build-native-ubuntu:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: recursive
+      
+    - name: Set up Rust
+      run: |
+        rustup default ${{ env.RUST_VERSION }}
+
+    - name: Build Rust Project
+      run: cargo build --manifest-path=./rust-bindings/Cargo.toml --release
+    
+    - name: Upload linux library
+      uses: actions/upload-artifact@master
+      with:
+        name: linux-library
+        path: ./rust-bindings/target/release/librust_bindings.so
+  
+  build-native-macos:
+    runs-on: macos-latest
```

Comment:
I had the same concern at it was difficult to find any documentation from Github where they specified the architecture used.
I could only find this on self-hosted runners, https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners, and some blogs. 
I agree it might change in the future so I have change to macos-12.